.github/workflows/lint.yml

```diff
@@ -0,0 +1,20 @@
+name: CI
+on: push
+jobs:
+  "Lint":
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-node@v4
+        with:
+          node-version: "20"
+          cache: "npm"
+      - name: Install modules
+        run: npm ci
+      - name: Run ESLint
+        run: npm run lint:report
+        continue-on-error: true
+      - name: Annotate Code Linting Results
+        uses: ataylorme/eslint-annotate-action@v3
+        with:
+          report-json: "eslint_report.json"
```

Comment:
This is the actual github action. This runs our lint:report step and then parses it using a 3rd party action that writes each error to github as an annotation. 

This means that you get lovely inline errors on your PR, showing you the problem and the failure:
<img width="1064" alt="image" src="https://github.com/user-attachments/assets/f4e718ff-32c9-4293-a066-070635e63308">